### PR TITLE
[SPARK-35813][SQL][DOCS] Add new adaptive config into sql-performance-tuning docs

### PR DESCRIPTION
**Pull request by ulysses-you**
### What changes were proposed in this pull request?

Add new configs in sql-performance-tuning docs.
* spark.sql.adaptive.coalescePartitions.parallelismFirst
* spark.sql.adaptive.coalescePartitions.minPartitionSize
* spark.sql.adaptive.autoBroadcastJoinThreshold
*  spark.sql.adaptive.maxShuffledHashJoinLocalMapThreshold

### Why are the changes needed?

Help user to find them.

### Does this PR introduce _any_ user-facing change?

yes, docs changed.

### How was this patch tested?

image

image

### CODE REVIEW

File: docs/sql-performance-tuning.md [hunk 1]
```diff
@@ -273,7 +273,32 @@ This feature coalesces the post shuffle partitions based on the map output stati
  </table>
  
 ### Converting sort-merge join to broadcast join
-AQE converts sort-merge join to broadcast hash join when the runtime statistics of any join side is smaller than the broadcast hash join threshold. This is not as efficient as planning a broadcast hash join in the first place, but it's better than keep doing the sort-merge join, as we can save the sorting of both the join sides, and read shuffle files locally to save network traffic(if `spark.sql.adaptive.localShuffleReader.enabled` is true)
+AQE converts sort-merge join to broadcast hash join when the runtime statistics of any join side is smaller than the adaptive broadcast hash join threshold. This is not as efficient as planning a broadcast hash join in the first place, but it's better than keep doing the sort-merge join, as we can save the sorting of both the join sides, and read shuffle files locally to save network traffic(if `spark.sql.adaptive.localShuffleReader.enabled` is true)
```
Comment (ulysses-you) [inline docs/sql-performance-tuning.md:276]: `smaller than the broadcast hash join`  ->  `smaller than the adaptive broadcast hash join`

**Comment by ulysses-you** [comment]
cc @cloud-fan @yaooqinn @HyukjinKwon

**Review by yaooqinn — APPROVED** [review]

File: docs/sql-performance-tuning.md [hunk 2]
```diff
@@ -273,7 +273,32 @@ This feature coalesces the post shuffle partitions based on the map output stati
...
+  <table class="table">
+     <tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr>
+     <tr>
+       <td><code>spark.sql.adaptive.autoBroadcastJoinThreshold</code></td>
+       <td>(none)</td>
```
Comment (cloud-fan) [inline docs/sql-performance-tuning.md:281]: shall we mention the default value is the same as `spark.sql.autoBroadcastJoinThreshold`?

File: docs/sql-performance-tuning.md [hunk 3]
```diff
@@ -273,7 +273,32 @@ This feature coalesces the post shuffle partitions based on the map output stati
...
+       <td>
+         Configures the maximum size in bytes for a table that will be broadcast to all worker nodes when performing a join. By setting this value to -1 broadcasting can be disabled. The default value is same with <code>spark.sql.autoBroadcastJoinThreshold</code>. Note that, this config is used only in adaptive framework.
+       </td>
+       <td>3.2.0</td>
+     </tr>
+  </table>
+
+### Converting sort-merge join to shuffled hash join
+AQE converts sort-merge join to shuffled hash join when all reduce partitions size are small enough, the max threshold can see the config `spark.sql.adaptive.maxShuffledHashJoinLocalMapThreshold`.
```
Comment (cloud-fan) [inline docs/sql-performance-tuning.md:290]: `... when all the post shuffle partitions are smaller than a threshold.`

**Comment by cloud-fan** [comment]
can we also update the `Coalescing Post Shuffle Partitions` section to replace the min partition num config with the new min partition size?

File: docs/sql-performance-tuning.md (hunk 2 above)
Comment (ulysses-you) [reply docs/sql-performance-tuning.md:281]: yee, mentioned it in `Meaning` description.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.2!

**Comment by cloud-fan** [comment]
missed one thing: `AQE is disabled by default.`

Can we correct this in the doc as well?

**Comment by ulysses-you** [comment]
thank you @cloud-fan , created https://github.com/apache/spark/pull/33295

**Pull request closed without merging**